QRCoder/Builders/Payloads/PayloadExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using QRCoder.Builders.Payloads;
+using QRCoder.Builders.Renderers;
+using QRCoder.Builders.Renderers.Implementations;
+
+namespace QRCoder
+{
+    public static class PayloadExtensions
+    {
+        public static T WithErrorCorrection<T>(this T payload, QRCodeGenerator.ECCLevel eccLevel)
+            where T : IConfigurableEccLevel
+        {
+            payload.EccLevel = eccLevel;
+            return payload;
+        }
+
+        public static T WithEciMode<T>(this T payload, QRCodeGenerator.EciMode eciMode)
+            where T : IConfigurableEciMode
+        {
+            payload.EciMode = eciMode;
+            return payload;
+        }
+
+        public static T WithVersion<T>(this T payload, int version)
+            where T : IConfigurableVersion
+        {
+            payload.Version = version;
+            return payload;
+        }
+
+        public static T RenderWith<T>(this IPayload payload)
+            where T : IRenderer, new()
+        {
+            var renderer = new T();
+            renderer.Payload = payload;
+            return renderer;
+        }
```

Comment:
This functions, but I think `RenderWithAsciiRenderer()` or `RenderAsPng()` would be better than generics here.  I specifically don't like that intellisense does not show the compatible renderers for `T`.  Sorta breaks the goal.

![image](https://github.com/codebude/QRCoder/assets/6377684/81209628-3196-49ae-875f-af45b32748c6)